onnxruntime/core/framework/cloud_invoker.cc

```diff
@@ -33,6 +36,175 @@ CloudEndPointInvoker::CloudEndPointInvoker(const CloudEndPointConfig& config,
   }
 }
 
+class CurlGlobal {
+ public:
+  static void Init() {
+    static CurlGlobal curl_global;
+  }
+
+ private:
+  CurlGlobal() {
+    // Thread-safety is a must since curl might also be initialized in triton client.
+    const auto* info = curl_version_info(CURLVERSION_NOW);
+    ORT_ENFORCE(info->features & CURL_VERSION_THREADSAFE, "curl global init not thread-safe, need to upgrade curl version!");
```

Comment:
Function local static variables are initialized in a thread-safe manner since C++11, so technically this is not needed but doesn't hurt to check.

Reply:
looks like we have to check it - https://curl.se/libcurl/c/libcurl.html

Reply:
But who will catch this exception? What would happen if here it throws?